### PR TITLE
Support AuthSource and AuthMechanism

### PR DESCRIPTION
**Pull request by duane-talentsonar**
Add support for the authSource and authMechanism URL options. As a side affect add support for connect options authenticationDatabase and authenticationMechanism.

### CODE REVIEW

**Comment by duane-talentsonar** [comment]
I will be forwarding this PR to the mongoctl.

File: mongoctl/commands/common/connect.py [hunk 1]
```diff
@@ -97,9 +98,12 @@ def open_mongo_shell_to_server(server,
         else:
             database = "admin"
 
-    if username or server.needs_to_auth(database):
+    login_database = database if not "authenticationDatabase" in shell_options \
```
Comment (mhfrantz) [inline mongoctl/commands/common/connect.py]: Where this idiom is used elsewhere, the default value comes after the `else`, so I think reversing what you have here is more consistent:

```
login_database = shell_options["authenticationDatabase"] if "authenticationDatabase" in shell_options else database
```

**Comment by mhfrantz** [comment]
FTP

**Comment by duane-talentsonar** [comment]
PTAL -- rebase

**Comment by mhfrantz** [comment]
LGTM rebase

**Comment by duane-talentsonar** [comment]
Replaced by #2.

**Pull request closed without merging**